### PR TITLE
Network december release --- DDoS policy customization

### PR DESCRIPTION
**Pull request by seyedfayaz**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json [hunk 1]
```diff
@@ -0,0 +1,291 @@
+{
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json]: Please put this file in `Microsoft.Network/stable/2018-10-01`, not `2018-08-01`

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json]: Please put this file in `Microsoft.Network/stable/2018-10-01`, not `2018-08-01` (same comment for other example jsons)

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/publicIpAddress.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "info": {
     "title": "NetworkManagementClient",
     "description": "The Microsoft Azure Network management API provides a RESTful set of web services that interact with Microsoft Azure Networks service to manage your network resources. The API has entities that capture the relationship between an end user and the Microsoft Azure Networks service.",
-    "version": "2018-08-01"
+    "version": "2018-10-01"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/publicIpAddress.json]: Please add those changes to `Microsoft.Network/stable/2018-10-01/publicIpAddress.json`, not here

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json (hunk 2 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/publicIpAddress.json (hunk 3 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/publicIpAddress.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+  "swagger": "2.0",
```
Comment (tejaskarandikar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json]: Can you remove these file from the PR?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+          "items": {
+            "type": "object",
+            "properties": {
+              "id":"string"
+            }
+          },
+          "description": "The list of public IPs associated with the DDoS custom policy resource. This list is read-only."
+        },
+        "protocolCustomSettings": {
+          "readOnly": true,
```
Comment (tejaskarandikar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json]: Why is protocolCustomSettings readonly? We need to let users set this in the PUT?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+          },
+          "description": "The list of public IPs associated with the DDoS custom policy resource. This list is read-only."
+        }
+      },
+      "description": "DDoS custom policy properties."
+    },
+    "ProtocolCustomSettingsFormat": {
+      "properties": {
+        "protocol": {
+          "readOnly": false,
```
Comment (tejaskarandikar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json]: Add enum for protocol please? Tcp, Udp and Syn

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/publicIpAddress.json [hunk 7]
```diff
@@ -452,6 +456,37 @@
...
+        "protectionCoverage": {
+          "readOnly": false,
+          "type": "string",
+          "enum": [
+            "Basic",
+            "Standard"            
+          ],
+          "description": "The type of the DDoS protection plan associated with the public IP."
+        },
+        "domainNameLabel": {
```
Comment (tejaskarandikar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/publicIpAddress.json]: Why is domainNameLabel, fqdn etc in the DDOS settings?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/examples/DdosCustomPolicyCreate.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
+{
+  "parameters": {
+    "api-version": "2018-10-01",
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg1",
+    "ddosCustomPolicyName": "test-ddos-custom-policy",
+    "parameters": {
+      "location": "westus",
+      "properties": {}
```
Comment (tejaskarandikar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/examples/DdosCustomPolicyCreate.json]: You need to provide atleast 1 protocol custom setting in the custom policy else the PUT is not allowed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/publicIpAddress.json [hunk 9]
```diff
@@ -358,6 +358,10 @@
           "$ref": "#/definitions/PublicIPAddressDnsSettings",
           "description": "The FQDN of the DNS record associated with the public IP address."
         },
+        "ddosSettings": {
+          "$ref": "#/definitions/DdoSSettings",
```
Comment (tejaskarandikar) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/publicIpAddress.json]: This should probably be ./networkInterface.json#/definitions/DdoSSettings, we should validate

**Review by tejaskarandikar — CHANGES_REQUESTED** [review]
Please take a look at the PR comments

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json (hunk 1 above)
Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json]: Don't see that it's done...

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json (hunk 2 above)
Comment (MikhailTryakhov) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json]: You should change folder where you put this file in the ```2018-10-01``` folder

**Review by MikhailTryakhov — CHANGES_REQUESTED** [review]
please move all the files in ```2018-10-01``` folder. now they're in ```2018-08-01```

**Comment by MikhailTryakhov** [comment]
@seyedfayaz there is no activity resolving comments provided, milestone is missed. please target on the next monthly branch (review milestones section in one note)

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/publicIpAddress.json (hunk 7 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/publicIpAddress.json]: removed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json (hunk 6 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json (hunk 5 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-10-01/ddosCustomPolicy.json]: fixed

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json (hunk 4 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json (hunk 2 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/examples/DdosCustomPolicyCreate.json]: done

File: specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json (hunk 1 above)
Comment (seyedfayaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/ddosCustomPolicy.json]: done now :)

**Comment by seyedfayaz** [comment]
creating a new pr

**Pull request closed without merging**